Adding TextToTermIndexSequenceTransform

turambar opened the pull request:
## What changes were proposed in this pull request?

Adding a TextToTermIndexSequenceTransform (similar to TextToCharacterIndexTransform but with terms).

## How was this patch tested?

Added a unit test to TestTransforms.

AlexDBlack commented on datavec-api/src/main/java/org/datavec/api/transform/transform/nlp/TextToTermIndexSequenceTransform.java:
```diff
@@ -0,0 +1,119 @@
+ * @author Dave Kale
+ */
+@Data
+@EqualsAndHashCode(callSuper = true, exclude = {""})
```
probably exclude={"writableMap"}

AlexDBlack commented on datavec-api/src/main/java/org/datavec/api/transform/transform/nlp/TextToTermIndexSequenceTransform.java:
```diff
@@ -0,0 +1,119 @@
+ */
+@Data
+@EqualsAndHashCode(callSuper = true, exclude = {""})
+@JsonIgnoreProperties({"wordIndexMap", "writableMap"})
```
Remove "wordIndexMap" otherwise your transform can't be (de)serialized from/to JSON (your vocab won't be present).

turambar commented:
@AlexDBlack addressed feedback on JSON annotations, and added a unit test to test serialization. However, the unit test fails because it looks like the call to equals is not properly ignoring the `writableMap` field as specified in the EqualsAndHashCode annotation. Any thoughts would be much appreciated!

turambar commented:
@AlexDBlack: I fixed the annotations on `BaseSequenceTransform` and the new `TextToTermIndexSequenceTransform` -- unit test now passes.
